community/kernel/src/main/java/org/neo4j/unsafe/batchinsert/BatchInserters.java

```diff
@@ -46,7 +47,7 @@
      * @param storeDir the store directory
      * @return a new {@link BatchInserter}
      */
-    public static BatchInserter inserter( String storeDir )
+    public static BatchInserter inserter( String storeDir ) throws IOException
```

Comment:
Good catch - breaking the API here isn't bad. I'll put a change together to recover that.

That said - it'll still get an `IOException`, just wrapped into a `RuntimeException`. From the client perspective, an IOException means what an IOException typically does: it couldn't write somewhere. In this case, it couldn't write to something in the specified `storeDir`. Client could handle it by reporting that a failure occurred accessing the specified directory - but reality is that they'll probably just let it die.